pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -671,6 +671,20 @@ func (c *Reconciler) reconcile(ctx context.Context, pr *v1.PipelineRun, getPipel
 	}
 
 	if pipelineRunFacts.State.IsBeforeFirstTaskRun() {
+		// check if the object's key names are referenced correctly i.e. param.objectParam.key1
```

Comment:
working on adding unit tests for this, same for taskrun. e2e tests are added to cover this code